plugins/hypervisors/xenserver/src/main/java/com/cloud/hypervisor/xenserver/resource/CitrixResourceBase.java

```diff
@@ -1522,12 +1522,16 @@ public void destroyPatchVbd(final Connection conn, final Set<VM> vms) throws Xml
                             vbd.insert(conn, patchVDI);
                             vbd.eject(conn);
                         }
+                    } catch (Exception e) {
+                        s_logger.debug("Cannot eject CD-ROM device for VM " + vmName + " due to " + e.toString(), e);
+                    }
+                    try {
                         vbd.destroy(conn);
-                        break;
+                    } catch (Exception e) {
+                        s_logger.debug("Cannot destroy CD-ROM device for VM " + vmName + " due to " + e.toString(), e);
```

Comment:
Same here.

Reply:
Ignorable errors, not logged to avoid confusion for admin/user. 99.99% we'll never hit that.